src/pudl_rmi/create_override_spreadsheets.py

```diff
@@ -0,0 +1,708 @@
+    "plant_name_ferc1",
+    "plant_name_new",
+    "fuel_type_code_pudl_ferc1",
+    "fuel_type_code_pudl_eia",
+    "net_generation_mwh_ferc1",
+    "net_generation_mwh_eia",
+    "capacity_mw_ferc1",
+    "capacity_mw_eia",
+    "capacity_factor_ferc1",
+    "capacity_factor_eia",
+    "total_fuel_cost_ferc1",
+    "total_fuel_cost_eia",
+    "total_mmbtu_ferc1",
+    "total_mmbtu_eia",
+    "fuel_cost_per_mmbtu_ferc1",
+    "fuel_cost_per_mmbtu_eia",
+    "installation_year_ferc1",
+    "installation_year_eia",
+]
+
+RELEVANT_COLS_PPL = [
+    "record_id_eia",
+    "report_year",
+    "utility_id_pudl",
+    "utility_id_eia",
+    "utility_name_eia",  # I add this in from the utils_eia860() table
+    "operational_status_pudl",
+    "true_gran",
+    "plant_part",
+    "ownership_dupe",
+    "fraction_owned",
+    "plant_id_eia",
+    "plant_id_pudl",
+    "plant_name_new",
+    "generator_id",
+    "capacity_mw",
+    "capacity_factor",
+    "net_generation_mwh",
+    "installation_year",
+    "fuel_type_code_pudl",
+    "total_fuel_cost",
+    "total_mmbtu",
+    "fuel_cost_per_mmbtu",
+    "heat_rate_mmbtu_mwh",
+]
+
+# --------------------------------------------------------------------------------------
+# Generate Override Tools
+# --------------------------------------------------------------------------------------
+
+
+def _pct_diff(df, col) -> None:
+    """Calculate percent difference between EIA and FERC versions of a column."""
+    df.loc[
+        (df[f"{col}_eia"] > 0) & (df[f"{col}_ferc1"] > 0), f"{col}_pct_diff"
+    ] = round(((df[f"{col}_ferc1"] - df[f"{col}_eia"]) / df[f"{col}_ferc1"] * 100), 2)
+
+
+def _is_best_match(df, cap_pct_diff=6, net_gen_pct_diff=6, inst_year_diff=3) -> None:
+    """Fill the best_match column with strings to show cap, net_gen, inst_year match.
+
+    The process of manually checking all of the FERC-EIA matches made by the machine
+    learning algorithm is tedius. This function makes it easier to speed through the
+    obviously good matches and pay more attention to those that are more questionable.
+
+    By default, a "best match" is comprised of a FERC-EIA match with a capacity percent
+    difference of less than 6%, a net generation percent difference of less than 6%,
+    and an installation year difference of less than 3 years.
+
+    """
+    best_cap = df.capacity_mw_pct_diff < cap_pct_diff
+    best_net_gen = df.net_generation_mwh_pct_diff < net_gen_pct_diff
+    best_inst_year = df.installation_year_diff < inst_year_diff
+
+    df.loc[:, "best_match"] = df.best_match.fillna("")
+    df.loc[best_cap, "best_match"] = "cap"
+    df.loc[best_net_gen, "best_match"] = df.best_match + "_net-gen"
+    df.loc[best_inst_year, "best_match"] = df.best_match + "_inst_year"
+    df.loc[:, "best_match"] = df.best_match.replace(r"^_", "", regex=True)
+
+
+def _prep_ferc_eia(ferc1_eia, pudl_out) -> pd.DataFrame:
+    """Prep FERC-EIA for use in override output sheet pre-utility subgroups."""
```

Comment:
this is just a suggestion so feel free to ignore but after looking at this function again, instead of worrying about the order throughout this function... it might be simpler to just explicitly set the order in one list and apply it at the end.

Reply:
What do you think about doing something similar to what you did in the Optimus Duke PR? Where you had a dict for renaming columns but it also included "fake" column names?

Reply:
i like that idea! i'd suggest swapping the keys and values of the dict so they can all just be `fake` or `null` or whatever. the order I used over there meant I needed to have `faked_{n}` which was easy to generate the first time but hard to update.

Reply:
I'll push what I have and you can see what you think!